agent/agent-gc-monitor/gc-monitor-core/src/main/java/com/microsoft/gcmonitor/notifications/NotificationObserver.java

```diff
@@ -55,19 +55,21 @@ public void handleNotification(@Nullable Notification notification, Object handb
   public void watchGcNotificationEvents() {
     executorService.submit(
         () -> {
-          //noinspection InfiniteLoopStatement
-          while (true) {
-            try {
-              NotificationJob sample = workQueue.poll(Long.MAX_VALUE, TimeUnit.SECONDS);
-              if (sample != null) {
-                sample.collector.update(sample.notification);
+          try {
+            //noinspection InfiniteLoopStatement
+            while (true) {
```

Comment:
I assume we've checked for possible memory leaks?

Reply:
Added a maximum capacity to prevent this